test-runner/src/reporters/base.ts

```diff
@@ -66,28 +65,24 @@ export class BaseReporter implements Reporter  {
   }
 
   onTestEnd(test: Test, result: TestResult) {
-    switch (result.status) {
-      case 'skipped': {
-        this.skipped.push(test);
-        return;
-      }
-      case 'passed':
-        if (test.results.length === 1)
-          this.passed.push(test);
-        else
-          this.flaky.push(test);
-        return;
-      case 'failed':
-        // Fall through.
-      case 'timedOut': {
-        if (test.results.length === this.config.retries + 1) {
-          if (result.status === 'timedOut')
-            this.timedOut.push(test);
-          else
-            this.failed.push(test);
-        }
-        return;
+    if (result.status === 'skipped') {
+      this.skipped.push(test);
+      return;
+    }
+
+    if (result.status === result.expectedStatus) {
+      if (test.results.length === 1) {
+        // as expected from the first attempt
+        this.asExpected.push(test);
+      } else {
+        // as expected after unexpected -> flaky.
+        this.flaky.push(test);
       }
+      return;
+    }
+    if (result.status === 'passed' || test.results.length === this.config.retries + 1) {
```

Comment:
This logic is brittle. Should we have a separate `test.retries` field that differs between tests, so that we only compare with expected number of retires?

Reply:
I don't think a separate retries fixes it because we are ok with any 'passed' :/